### ISSUE TITLE
Filtering the standard logs which need to be sent to the loguru

### ISSUE THREAD

**Issue by Delgan** [Delgan/loguru#926]
Thank you so much for your help. Initially, I used bind() with filter but it clubs output into single file.
When I tried to route the output of other logger to loguru, it is displaying so many internal logs. Can we filter the standard logs which need to be sent to the loguru.
My code for same is as below.

```
def get_logger():
    global _logger
    if _logger is None:
        class InterceptHandler(logging.Handler):
            def emit(self, record):
                # Get corresponding Loguru level if it exists.
                try:
                    level = logger.level(record.levelname).name
                except ValueError:
                    level = record.levelno

                # Find caller from where originated the logged message.
                frame, depth = sys._getframe(6), 6
                while frame and frame.f_code.co_filename == logging.__file__:
                    frame = frame.f_back
                    depth += 1

                logger.opt(depth=depth, exception=record.exc_info).log("INFO", record.getMessage())

        logging.basicConfig(handlers=[InterceptHandler()], level="INFO", force=True)
        _logger = lg.get_app_logger(".", LOG_FILE_PATH)

    return _logger
```

_Originally posted by @durgeksh in https://github.com/Delgan/loguru/issues/25#issuecomment-1639508232_

**Comment by Delgan** [comment]
@durgeksh I opened a new ticket regarding the difficulties you're facing.

What do you mean by "internal logs"? Do you have any example? Did you try to use a `filter` maybe?

**Comment by durgeksh** [comment]
Internal logs means the logs propagated by the other modules.

**Comment by Delgan** [comment]
Can you use a `filter` and ignore messages from specific modules?

**Comment by durgeksh** [comment]
> Can you use a `filter` and ignore messages from specific modules?

but, it still gets into the log file which I have kept as program.log.

**Comment by Delgan** [comment]
Can you summarize what is the expected output you're trying to achieve please?

**Comment by Delgan** [comment]
It's hard to tell where the problem comes from because you're configuration is quite complex.

Instead of thinking in terms of "logger", I'd recommend thinking in terms of "handler". With that principle, I would rephrase your requirements as follows:
* There is one console handler that must logs everything but the third-party library logs
* There is one file handler per task logging the task itself and third-party library logs

Now, how do we translate this into a Loguru configuration?

Let's first install the `InterceptHandler` to make sure logs from the third-party library are available:

```python
class InterceptHandler(logging.Handler):
    def emit(self, record):
        # Get corresponding Loguru level if it exists.
        try:
            level = logger.level(record.levelname).name
        except ValueError:
            level = record.levelno

        # Find caller from where originated the logged message.
        frame, depth = sys._getframe(6), 6
        while frame and frame.f_code.co_filename == logging.__file__:
            frame = frame.f_back
            depth += 1

        logger.opt(depth=depth, exception=record.exc_info).log(level, record.getMessage())

logging.basicConfig(handlers=[InterceptHandler()], level=0, force=True)
```

Now all handlers will receive third-party logs, unless they are explicitly filtered out.

We can easily configure the console handler we defined:

```python
logger.add(sys.stderr, filter={"pika.adapters": False})
```

Regarding the task handlers, we can use a similar `filter` which accepts a specific task logs or `pika.adpaters` logs:

```python
def task_filter(record, task_id):
    return record["extra"].get("task_id") == task_id or (record["name"] + ".").startswith("pika.adapters.")

logger.add(f"task_{task_id}.log", filter=lambda record: task_filter(record, task_id))
```

Finally, we can build it all in your `main.py`:

```python
from loguru import logger

def task_filter(record, task_id):
    return record["extra"].get("task_id") == task_id or (record["name"] + ".").startswith("pika.adapters.")

def start_task(task_id):
    task_handler = logger.add(f"task_{task_id}.log", filter=lambda record: task_filter(record, task_id))
    try:
        with logger.contextualize(task_id=task_id):
            logger.info("Task started")
            call_another_function()
    finally:
        logger.remove(task_handler)

@logger.catch(reraise=True)
def test():
    logger.remove()
    logger.add(sys.stderr, filter={"pika.adapters": False})

    app_logger.info("App started")
    start_task(42)

if __name__ == "__main__":
    test()
```

**Comment by durgeksh** [comment]
Thank you for your reply. I will make changes according to code suggested by you and update you.

**Comment by durgeksh** [comment]
Thank you. is there any other way to not print the statements into logfile for one logger instance and in other instance print it to log file.

**Comment by Delgan** [comment]
> Thank you. is there any other way to not print the statements into logfile for one logger instance and in other instance print it to log file.

Not right now, although I'm planning to add `logger.new()` (to help create independent loggers) and `logger.bridge()` (to help interoperate with standard logging) methods at some point.
